File: 0190-reverse-bits/0190-reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        ans = 0
        for i in range(32):
            if (n>>i)&1:
                ans |= 1<<(31-i)
        return ans

'''
    00000010100101000001111010011100

    00111001011110000010100101000000
'''                

File: 0190-reverse-bits/0190-reverse-bits.cpp
class Solution {
public:
    int reverseBits(int n) {
        
       // signed integer
       // 1bit sign
       uint32_t nn = static_cast<uint32_t>(n);
       uint32_t reversed = 0;
       for(int i=0; i<32; i++) {
            if(n&1) {
                reversed |= (1<<(31-i));
            }
            n>>=1;
       }
       return static_cast<int>(reversed);
    }
};